### PR TITLE
Add managers to blade and enclosure

### LINKED ISSUE #54: Implement Manager for Enclosure and Blade
As oneview-redfish-toolkit, I access "/redfish/v1/Manager/<server_hardware_uuid>" or "/redfish/v1/Manager/<enclosure_id>" and receive basic information respectively on the server hardware and enclosure managers containing all mandatory fields and links to related Chassis and Compute Systems.

### PR DESCRIPTION
**Pull request by ffdarkpenguin**
Closes #54 

---
This change is [Reviewable](https://reviewable.io/reviews/hewlettpackard/oneview-redfish-toolkit/84)

### CODE REVIEW

File: oneview_redfish_toolkit/api/blade_manager.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class BladeManager(Manager):
+    """Creates a Manager Redfish dict for a Blade
+
+        Populates self.redfish with some hardcoded Manager
+        values and with the response from OneView.
+    """
+
+    def __init__(self, server_hardware, ov_version):
```
Comment (llvieira) [inline oneview_redfish_toolkit/api/blade_manager.py]: Add description for the parameter **ov_version** on constructor documentation.

File: oneview_redfish_toolkit/api/enclosure_manager.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class EnclosureManager(Manager):
+    """Creates a Manager Redfish dict for Enclosure
+
+        Populates self.redfish with some hardcoded Manager
+        values and with the response from OneView.
+    """
+
+    def __init__(self, enclosure, ov_version):
```
Comment (llvieira) [inline oneview_redfish_toolkit/api/enclosure_manager.py]: Description for **ov_version**

File: oneview_redfish_toolkit/api/manager.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+class Manager(RedfishJsonValidator):
+    """Creates a Manager Redfish dict
+
+        Populates self.redfish with some hardcoded Manager
+        values and with the response from OneView.
+    """
+
+    SCHEMA_NAME = 'Manager'
+
+    def __init__(self, oneview_resource, ov_version):
```
Comment (llvieira) [inline oneview_redfish_toolkit/api/manager.py]: Description for **ov_version**

**Comment by llvieira** [comment]
Reviewed 10 of 10 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84)*

**Comment by danielreed** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[oneview_redfish_toolkit/api/manager.py, line 45 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-Kuz6TtnLJ2qjmUiXdXo:-Kuz6TtnLJ2qjmUiXdXp:b-scofyn) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L45)):*
> ```Python
>         self.redfish["@odata.type"] = "#Manager.v1_3_1.Manager"
>         self.redfish["Id"] = oneview_resource['uuid']
>         self.redfish["Name"] = "Synergy"
> ```

Should the names be something like 'Blade Manager' or 'Enclosure Manager', instead of Synergy?

---

*[oneview_redfish_toolkit/api/manager.py, line 50 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-Kuz6B-nGUxFyfhJDp3B:-Kuz6B-nGUxFyfhJDp3C:b-gw1jm4) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L50)):*
> ```Python
>         self.redfish["Status"] = collections.OrderedDict()
>         self.redfish["Status"]["State"] = "Enabled"
>         self.redfish["Status"]["Health"] = oneview_resource["status"]
> ```

duplicate lines here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84)*

**Comment by ffdarkpenguin** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[oneview_redfish_toolkit/api/blade_manager.py, line 29 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-KuyzIdO9-72mSpEVDEy-r1-29:-Kuzp4v7b1IQY8SDlxNo:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/blade_manager.py#L29)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Add description for the parameter **ov_version** on constructor documentation.
</blockquote></details>

Done.

---

*[oneview_redfish_toolkit/api/enclosure_manager.py, line 29 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-KuyzIdO9-72mSpEVDEz-r1-29:-Kuzp4HhZ1loLfgijBpY:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/enclosure_manager.py#L29)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Description for **ov_version**
</blockquote></details>

Done.

---

*[oneview_redfish_toolkit/api/manager.py, line 31 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-KuyzIdO9-72mSpEVDF--r1-31:-Kuzp3inujpc8VDgYohc:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L31)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Description for **ov_version**
</blockquote></details>

Done.

---

*[oneview_redfish_toolkit/api/manager.py, line 45 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-Kuz6TtnLJ2qjmUiXdXo:-Kuzp20wdONd7VlPXRyK:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L45)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

Should the names be something like 'Blade Manager' or 'Enclosure Manager', instead of Synergy?
</blockquote></details>

Done.

---

*[oneview_redfish_toolkit/api/manager.py, line 50 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-Kuz6B-nGUxFyfhJDp3B:-Kuzp2V7BgelFtNAqMMU:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L50)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

duplicate lines here?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84)*

**Comment by ricardoas** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[oneview_redfish_toolkit/api/manager.py, line 49 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-Kv2aoKBbtsINX-DAcDc:-Kv2aoKBbtsINX-DAcDd:b-84we8b) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L49)):*
> ```Python
>         self.redfish["FirmwareVersion"] = ov_version
>         self.redfish["Status"] = collections.OrderedDict()
>         self.redfish["Status"]["State"] = "Enabled"
> ```

Are we mapping all resources status like this? If so we have so further improvements:
* there are a few possible values for redfish State, we should at least be using `Disabled` then `oneview_resource["status"]` is `Disabled`
* redfish Health possible values are: "OK", "Warning", "Critical" and null according to the schema... we need to work on the case when`oneview_resource["status"]` equals `Disabled` (State == `Disabled`? ) and `Unknown` (State == `Absent`?), not just assign it directly.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84)*

**Comment by ricardoas** [comment]
We also need some links from System and Chassis to their managers... better if a add a new card for this.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-:-Kv2edaxhqv3D1sYd1yp:b-w3ybxz)*

**Comment by ffdarkpenguin** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[oneview_redfish_toolkit/api/manager.py, line 49 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-Kv2aoKBbtsINX-DAcDc:-Kv2vq-PHYi10eqgPnkd:b-gekho2) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/9a7fd819051910725962fb256309bf6fea943cd3/oneview_redfish_toolkit/api/manager.py#L49)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

Are we mapping all resources status like this? If so we have so further improvements:
* there are a few possible values for redfish State, we should at least be using `Disabled` then `oneview_resource["status"]` is `Disabled`
* redfish Health possible values are: "OK", "Warning", "Critical" and null according to the schema... we need to work on the case when`oneview_resource["status"]` equals `Disabled` (State == `Disabled`? ) and `Unknown` (State == `Absent`?), not just assign it directly.
</blockquote></details>

Sometime ago someone told to use all like this in a debate of our.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84)*

**Comment by danielreed** [comment]
:lgtm:

---

Review status: 5 of 10 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-:-Kv314nC1zamqssT7qWo:bnfp4nl)*

**Comment by ricardoas** [comment]
:lgtm:

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/84#-:-Kv33bPgowumVzDZra3w:bnfp4nl)*

**Pull request merged**